pkg/webhookreceiver/handler.go

```diff
@@ -0,0 +1,32 @@
+package webhookreceiver
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/prometheus/alertmanager/template"
+)
+
+type Handler struct {
+	// Route to set method, path etc.
+	Route func(r *mux.Route)
+
+	// HTTP Handler
+	Func http.HandlerFunc
+}
+
+// AddRoute adds the handler's route the to the router.
+func (h Handler) AddRoute(r *mux.Router) {
+	h.Route(r.NewRoute().HandlerFunc(h.Func))
+}
+
+// Alert Manager receiver response
+type AMReceiverResponse struct {
+	Status string
+}
+
+// Set webhook receiver path
+const AMReceiverPath = "/alertmanager-receiver"
```

Comment:
```suggestion
const AMReceiverPath = "/receiver"
```
Should we simplify the URL path without hyphens? Reference [link](https://swagger.io/resources/articles/best-practices-in-api-design/#nouns-describe-urls-better-3)

Reply:
If there are going to be other receivers in future too then makes sense to have "alertmanager" in path.. But have rarely seen any API URL path with "hyphen" in it.

Reply:
The reason for choosing this path was because this is what was used for POC, additionally there's some anecdotal evidence that hyphens is preferred way to separate the verbs in rest api path. I personally like the name as it identifies path as alertmanager

Reply:
Agreed.. We can revisit this in future when required too.. For now, resolving the comment. Thanks Tomas!